### PR TITLE
fix tag splitting for 3.X-upcoming repos (SOFTWARE-4475)

### PR DESCRIPTION
**Pull request by edquist**
Sorry, this one slid under my radar since it didn't actually have the `upcoming` keyword anywhere near it

### CODE REVIEW

**Comment by matyasselmeci** [comment]
What's going to happen when this encounters one of the old upcoming tags?

**Comment by edquist** [comment]
Ah.  Well, in that case something slightly broken i guess.  (You'll get `osg-upcoming` instead of just `upcoming`.)

But once the osg-tags are refreshed (when update_mashfiles.sh is run, which we can do manually when we upgrade or else it will run by itself nightly), then it will not encounter any of the old upcoming tags, and the situation will correct itself.

File: bin/update_mirror.py [hunk 1]
```diff
@@ -56,8 +56,16 @@ def lock(path):
 if socket.gethostname() == "repo-itb.opensciencegrid.org":
     hostname="repo-itb.opensciencegrid.org"
 
+def tagsplit(tag):
+    if 'upcoming' in tag:
```
Comment (edquist) [inline bin/update_mirror.py:60]: We can do it like this to be more explicit, if you like..?

```suggestion
    if 'upcoming' in tag and tag.startswith("osg-3."):
```

Comment (matyasselmeci) [reply bin/update_mirror.py:60]: yeah, please.

**Review by matyasselmeci — APPROVED** [review]
other than the one request, this is fine so pre-approving

**Pull request merged**